### PR TITLE
Control server in state machine

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: statemachine-control/src/main/scala/fluence/statemachine/control/ControlSignals.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    } yield ()
+
+  val changePeers: Resource[F, List[ChangePeer]] =
+    Resource.make(changePeersRef.tryTake.map(_.toList.flatten))(_ => changePeersRef.tryPut(Nil).void)
+
+  val stop: F[Unit] =
+    stopRef.take
+}
+
+sealed trait ControlSignal
```
Comment (alari) [inline statemachine-control/src/main/scala/fluence/statemachine/control/ControlSignals.scala]: Please move it to separate file

File: node/src/main/scala/fluence/node/eth/NodeEth.scala [hunk 2]
```diff
@@ -42,7 +42,7 @@ trait NodeEth[F[_]] {
   def nodeEvents: fs2.Stream[F, NodeEthEvent]
 
   /**
-   * Returns the expected node state, how it's built with received Ethereum data
+   * Returns the expected node state, how it's builbuilt with received Ethereum data
```
Comment (DieMyst) [inline node/src/main/scala/fluence/node/eth/NodeEth.scala]: `builbuilt`?

File: statemachine/control/src/main/scala/fluence/statemachine/control/ControlSignal.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package fluence.statemachine.control
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import scodec.bits.ByteVector
+
+sealed trait ControlSignal
+// A signal to change a voting power of the specified Tendermint validator. Voting power zero votes to remove.
```
Comment (DieMyst) [inline statemachine/control/src/main/scala/fluence/statemachine/control/ControlSignal.scala]: write docs in correct format 
```
/**
  *
  */
```

File: statemachine/control/src/main/scala/fluence/statemachine/control/ControlSignal.scala [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package fluence.statemachine.control
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import scodec.bits.ByteVector
+
+sealed trait ControlSignal
```
Comment (DieMyst) [inline statemachine/control/src/main/scala/fluence/statemachine/control/ControlSignal.scala]: docs about `ControlSignal` generally

File: statemachine/src/main/scala/fluence/statemachine/AbciHandler.scala [hunk 5]
```diff
@@ -177,26 +177,31 @@ class AbciHandler(
...
-          )
-      })
-      .map(_.build())
+      .use(
+        IO.pure(_)
+          .map(_.foldLeft(ResponseEndBlock.newBuilder()) {
+            case (resp, ChangePeer(keyType, validatorKey, votePower)) ⇒
+              resp
+                .addValidatorUpdates(
+                  ValidatorUpdate
```
Comment (DieMyst) [inline statemachine/src/main/scala/fluence/statemachine/AbciHandler.scala]: very nested method, split it in several methods

File: statemachine/src/main/scala/fluence/statemachine/config/StateMachineConfig.scala [hunk 6]
```diff
@@ -36,7 +37,14 @@ import scala.language.higherKinds
  * @param moduleFiles sequence of files with WASM module code
  * @param logLevel level of logging ( OFF / ERROR / WARN / INFO / DEBUG / TRACE )
  */
-case class StateMachineConfig(sessionExpirationPeriod: Long, moduleFiles: List[String], logLevel: String) {
+case class StateMachineConfig(
+  sessionExpirationPeriod: Long,
+  moduleFiles: List[String],
+  logLevel: String,
+  metricsPort: Short,
```
Comment (DieMyst) [inline statemachine/src/main/scala/fluence/statemachine/config/StateMachineConfig.scala]: metricPort, abciPort and control in docs

**Pull request merged**